backend/api/professor/grading_materials.py

```
@@ -0,0 +1,333 @@
+        elif file_extension == '.pdf':
+            # For PDF files, we'll use a simple text extraction
+            # In production, you might want to use PyPDF2 or pdfplumber
+            try:
+                import PyPDF2
+                import io
+                
+                pdf_reader = PyPDF2.PdfReader(io.BytesIO(content))
+                text = ""
+                for page in pdf_reader.pages:
+                    text += page.extract_text() + "\n"
+                return text
```

Ruff (0.14.1)

Consider moving this statement to an `else` block

(TRY300)